src/js/crypto.js

```diff
@@ -13,7 +13,74 @@
  * limitations under the License.
  */
 
-var hashes = ['sha1'];
+var shaTypes = {
+  'sha1': 4,
+  'sha256': 6,
+};
+
+var hashes = ['sha1', 'sha256'];
+
+function Verify(signtype) {
+  if (!(this instanceof Verify)) {
+    return new Verify(signtype);
+  }
+
+  signtype = signtype.toLowerCase();
+
+  if (hashes.indexOf(signtype) < 0) {
+    throw new Error('Unknown signing algorithm.' +
+                    'Please use crypto.getSignatures()');
+  }
+
+  Object.defineProperty(this, 'hashtype', {
+    value: signtype,
+    writable: false,
```

Comment:
add `configurable:false`. It is the default but it is good to emphise it. Or remove `writable:false`, since that is default as well.

Maybe just add these two as a comment. Less byte code size, but the reader knows the intention.